test: Skip doc generation when the template file is not found.

t2gran opened the pull request:
### Summary

When the unit-tests are run using the jar file, not a checked out project, then the doc folder might not be present. In this case we want to skip the doc generation, and not fail.

When fixing this I considered 3 different ways to do it:
 1. Use JUnit tags, and exclude tests using Maven profile
 2. Use annotation to include/exclude test
 3. Check if the docs folder exist, if not exit test

I went for option 3 - the least complex one. It has the downside that you need to include
```
    if(skipDocGeneration(BuildConfigurationDocTest.class)) {
      return;
    }
```
in every doc test (currently 5 places), but this is just slightly more code than using an JUnit annotation. There is no Maven config involved or build CI environment setup/changes needed.

optionsome reviewed (commented):
I quickly checked this and it looked otherwise ok but in our CI there were also example config tests failing with the following stack trace:
```
2022-11-01T14:34:04.3678026Z [ERROR] org.opentripplanner.standalone.config.ExampleConfigTest.routerConfig(Path)  Time elapsed: 0.257 s  <<< ERROR!
2022-11-01T14:34:04.3679216Z org.junit.platform.commons.PreconditionViolationException: Configuration error: You must configure at least one set of arguments for this @ParameterizedTest
2022-11-01T14:34:04.3680515Z 	at org.junit.platform.commons.util.Preconditions.condition(Preconditions.java:281)
2022-11-01T14:34:04.3681321Z 	at org.junit.jupiter.params.ParameterizedTestExtension.lambda$provideTestTemplateInvocationContexts$5(ParameterizedTestExtension.java:98)
2022-11-01T14:34:04.3682058Z 	at java.base/java.util.stream.AbstractPipeline.close(AbstractPipeline.java:323)
2022-11-01T14:34:04.3682720Z 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:283)
2022-11-01T14:34:04.3683298Z 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)

```

leonardehrenfried commented:
Then you need to disable this test. It looks for files but since you don't have the docs folder it cannot find any and fails the test.

optionsome commented:
```
2022-11-01T14:34:04.3678026Z [ERROR] org.opentripplanner.standalone.config.ExampleConfigTest.routerConfig(Path)  Time elapsed: 0.257 s  <<< ERROR!
2022-11-01T14:34:04.3679216Z org.junit.platform.commons.PreconditionViolationException: Configuration error: You must configure at least one set of arguments for this @ParameterizedTest
2022-11-01T14:34:04.3680515Z 	at org.junit.platform.commons.util.Preconditions.condition(Preconditions.java:281)
2022-11-01T14:34:04.3681321Z 	at org.junit.jupiter.params.ParameterizedTestExtension.lambda$provideTestTemplateInvocationContexts$5(ParameterizedTestExtension.java:98)
2022-11-01T14:34:04.3682058Z 	at java.base/java.util.stream.AbstractPipeline.close(AbstractPipeline.java:323)
2022-11-01T14:34:04.3682720Z 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:283)
2022-11-01T14:34:04.3683298Z 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
2022-11-01T14:34:04.3683883Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
2022-11-01T14:34:04.3685064Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2022-11-01T14:34:04.3685697Z 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
2022-11-01T14:34:04.3686268Z 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
2022-11-01T14:34:04.3687259Z 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2022-11-01T14:34:04.3687836Z 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
2022-11-01T14:34:04.3688561Z 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:107)
2022-11-01T14:34:04.3689396Z 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:42)
2022-11-01T14:34:04.3690222Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
2022-11-01T14:34:04.3691027Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3691840Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-11-01T14:34:04.3692537Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-11-01T14:34:04.3693236Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-11-01T14:34:04.3694014Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3694840Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-11-01T14:34:04.3695610Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-11-01T14:34:04.3696186Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2022-11-01T14:34:04.3697051Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
2022-11-01T14:34:04.3698073Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2022-11-01T14:34:04.3698879Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3699664Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-11-01T14:34:04.3700355Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-11-01T14:34:04.3701052Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-11-01T14:34:04.3701841Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3702660Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-11-01T14:34:04.3703432Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-11-01T14:34:04.3704013Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2022-11-01T14:34:04.3705046Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
2022-11-01T14:34:04.3706084Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2022-11-01T14:34:04.3706950Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3707742Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-11-01T14:34:04.3708434Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-11-01T14:34:04.3709124Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-11-01T14:34:04.3709907Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3710720Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-11-01T14:34:04.3711494Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-11-01T14:34:04.3712479Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
2022-11-01T14:34:04.3713553Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2022-11-01T14:34:04.3714449Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
2022-11-01T14:34:04.3715426Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
2022-11-01T14:34:04.3716236Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
2022-11-01T14:34:04.3717048Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
2022-11-01T14:34:04.3717952Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
2022-11-01T14:34:04.3718842Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
2022-11-01T14:34:04.3719566Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
2022-11-01T14:34:04.3720197Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
2022-11-01T14:34:04.3720908Z 	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
2022-11-01T14:34:04.3721780Z 	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
2022-11-01T14:34:04.3722574Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
2022-11-01T14:34:04.3723226Z 	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
2022-11-01T14:34:04.3723874Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
2022-11-01T14:34:04.3724710Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
2022-11-01T14:34:04.3725544Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
2022-11-01T14:34:04.3726284Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
2022-11-01T14:34:04.3726934Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2022-11-01T14:34:04.3727494Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
2022-11-01T14:34:04.3728059Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
2022-11-01T14:34:04.3729343Z 
2022-11-01T14:34:04.3729847Z [ERROR] org.opentripplanner.standalone.config.ExampleConfigTest.buildConfig(Path)  Time elapsed: 0.494 s  <<< ERROR!
2022-11-01T14:34:04.3730677Z org.junit.platform.commons.PreconditionViolationException: Configuration error: You must configure at least one set of arguments for this @ParameterizedTest
2022-11-01T14:34:04.3731482Z 	at org.junit.platform.commons.util.Preconditions.condition(Preconditions.java:281)
2022-11-01T14:34:04.3732271Z 	at org.junit.jupiter.params.ParameterizedTestExtension.lambda$provideTestTemplateInvocationContexts$5(ParameterizedTestExtension.java:98)
2022-11-01T14:34:04.3733001Z 	at java.base/java.util.stream.AbstractPipeline.close(AbstractPipeline.java:323)
2022-11-01T14:34:04.3733554Z 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:283)
2022-11-01T14:34:04.3734121Z 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
2022-11-01T14:34:04.3734686Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
2022-11-01T14:34:04.3735289Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2022-11-01T14:34:04.3735884Z 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
2022-11-01T14:34:04.3736460Z 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
2022-11-01T14:34:04.3737031Z 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2022-11-01T14:34:04.3737597Z 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
2022-11-01T14:34:04.3738312Z 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:107)
2022-11-01T14:34:04.3739137Z 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:42)
2022-11-01T14:34:04.3739948Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
2022-11-01T14:34:04.3740864Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3741564Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-11-01T14:34:04.3742171Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-11-01T14:34:04.3742778Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-11-01T14:34:04.3743469Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3744174Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-11-01T14:34:04.3744851Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-11-01T14:34:04.3745357Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2022-11-01T14:34:04.3746108Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
2022-11-01T14:34:04.3746997Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2022-11-01T14:34:04.3747692Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3748384Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-11-01T14:34:04.3748966Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-11-01T14:34:04.3749569Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-11-01T14:34:04.3750254Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3751021Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-11-01T14:34:04.3751697Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-11-01T14:34:04.3752203Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2022-11-01T14:34:04.3753000Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
2022-11-01T14:34:04.3753881Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2022-11-01T14:34:04.3782971Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3783808Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-11-01T14:34:04.3784435Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-11-01T14:34:04.3785065Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-11-01T14:34:04.3785766Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-11-01T14:34:04.3786503Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-11-01T14:34:04.3787173Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-11-01T14:34:04.3788028Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
2022-11-01T14:34:04.3788955Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2022-11-01T14:34:04.3789733Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
2022-11-01T14:34:04.3790477Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
2022-11-01T14:34:04.3791187Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
2022-11-01T14:34:04.3791903Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
2022-11-01T14:34:04.3792680Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
2022-11-01T14:34:04.3793451Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
2022-11-01T14:34:04.3794074Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
2022-11-01T14:34:04.3794841Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
2022-11-01T14:34:04.3795476Z 	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
2022-11-01T14:34:04.3796166Z 	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
2022-11-01T14:34:04.3796872Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
2022-11-01T14:34:04.3797433Z 	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
2022-11-01T14:34:04.3798001Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
2022-11-01T14:34:04.3798733Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
2022-11-01T14:34:04.3799455Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
2022-11-01T14:34:04.3800100Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
2022-11-01T14:34:04.3800886Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2022-11-01T14:34:04.3801396Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
2022-11-01T14:34:04.3801881Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
2022-11-01T14:34:04.3802235Z 
```

t2gran commented:
Since the `ExampleConfigTest` is annotation based(parameters injection) I had to use the `@EnabledIf`. I did it on the class level - witch disable ALL tests and the instantiation of the test if the `/docs` directory does not exist (and logs a warning).

So, I switched to strategy number 2 (PR Summary). Just add the new `@OnlyIfDocsExist` annotation to your test class, and it is skipped if the environment is not the standard. This can happen in many ways. I have encountered this if tests are run with different working directory(this is how I tested the feature), using jar files, or Maven multi-module project.

@optionsome Tip! I added this little trick around your very-long-stack-strace, then you can expand/collapse it:
```
<details>
  <summary>Stacktrace</summary>
 :
</details>
```
